### ISSUE TITLE
Bool switch

### ISSUE THREAD

**Issue by Campbellony** [commandlineparser/commandline#253]
My desired solution would enable the user to include an argument if they want the application automated, or omit the argument if they want to see an user interface.  In the example below, the Automate property does not get set to true when the parameter is specified.
`    class Options
    {
        [Option('a', "automate", Required = false, Default = false,
          HelpText = "Run in batch mode without user inteface.")]
        public bool Automate { get; set; }
    }`

**Comment by nemec** [comment]
What version of the library are you using? What arguments are you passing to the parser? It works for me with the behavior you expect.

    Parser.Default.ParseArguments<Options>("--automate".Split())  // Sets true

    Parser.Default.ParseArguments<Options>("".Split())  // Sets false

**Comment by Campbellony** [comment]
I'm using 2.2.1.  I see my problem now, I'm not setting my options class to the result of the parse.  I'm a bit confused how to do so now.  The last time I used this, I believe I passed the options into the parser, and it returned true/false if the parse succeeded.

**Comment by nemec** [comment]
```csharp
var result = Parser.Default.ParseArguments<Options>("--automate".Split());

result.WithParsed(options => /* do something with options*/);
result.WithNotParsed(IEnumerable<Error> errs => /* do something with error messages */);
```

**Comment by Campbellony** [comment]
Thanks for your help, I'm pretty new to functional style programming.  I've added:
`var result = CommandLine.Parser.Default.ParseArguments<Options>(args);
result.WithParsed(options => OnSuccessfulParse(options)) // options is an instance of Options type
result.WithNotParsed(errors => OnFailedParse(errors)); // errors is a sequence of type IEnumerable<Error>`

I created two methods for handling successful parse and failed parse.  However I'm getting the following error.

> 'ParserResult<Options>' does not contain a definition for 'WithParsed' and no extension method 'WithParsed' accepting a first argument of type 'ParserResult<Options>' could be found (are you missing a using directive or an assembly reference?)
